lib/xchain/abi.go

```diff
@@ -147,14 +154,20 @@ func SubmissionFromBinding(sub bindings.XSubmission, destChainID uint64) (Submis
 }
 
 // SubmissionToBinding converts a go xchain submission to a solidity binding submission.
-func SubmissionToBinding(sub Submission) bindings.XSubmission {
+func SubmissionToBinding(sub Submission) (bindings.XSubmission, error) {
```

Comment:
the best place to verify data is when you get it over the wire (or any IO).

a submission is a in-memory calculated type. Rather very the attestation we get from cprovider in `Worker::newCallback` via something like: `att.Verify()`.

Reply:
https://github.com/omni-network/omni/pull/2137/commits/2a5e8b3dc89eee068f432c54e1de4f66ed932cb2